tools/autograd/gen_variable_type.py

```diff
@@ -586,14 +604,31 @@ def emit_call(env):
         combined = nested_dict(env, declaration)
         extra_wrapping_stmts = []
         if strategy == 'use_derived':
-            call = CALL_VIA_DERIVED.substitute(combined)
-            if not modifies_arguments:
-                call, extra_wrapping_stmts = wrap_output(call)
+            # We only care about adding `at::AutoNonVariableTypeMode` guard for `baseType` dispatch
+            # (which corresponds to 'use_derived' strategy). The purpose of this guard is to make sure
+            # the baseType operations still dispatch to non-Variable type, even if the arguments passed
+            # in are now Variables.
+            # See NOTE [ Treating Variables as non-Variables in type dispatch ] for details.
+            code_block = ''
+            base_type_call = CALL_VIA_DERIVED.substitute(combined)
+            if not modifies_arguments and not returns_void:
+                if not modifies_arguments:
+                    rhs_value, extra_wrapping_stmts = wrap_output("tmp")
+                else:
+                    rhs_value = "tmp"
+                code_block = DISPATCH_TO_NON_VAR_TYPE_WITH_RETURN_VALUES.substitute(
```

Comment:
this `call = code_block` doesn't really do anything.  Is the objection to calling it `call` (which is done anyway)?